### DISCUSSION TITLE
Please help me with train text from scratch

### DISCUSSION THREAD

**Discussion by lbarasc** [ggml-org/llama.cpp#11023] [Q&A]
hi,
I really want to create mini LLM from scratch with Llamacpp.
Please help me how to use train-text-from-scratch or llama-train-text-from-scratch with text raw text. (like a public domain book for example) and generate logical text !
I test several configuration but the results are not really good.
what are the parameters you use and how long does it take (with cpu 24 threads) or gpu.
thank you for all.

**Comment by nvrxq** [comment]
Hi.
On older revisions you can find the examples/train-text-from-scratch folder, but I don't see it there now. You can read why it was removed. 
#8669

**Comment by lbarasc** [comment]
ok, thank you, i will read this and i hope i will find some trick to correctly train text from scratch
